File: hackpdm_odoo/hackpdm/custom_addons/custom_module/models/hp_category.py
from odoo import models, fields, api

class hp_category(models.Model):
    #base fields
    _name = 'hp.category'
    _description = 'category'
    _inherit = 'hp.common.model'

    #fields
    name = fields.Char(
        string='category name', 
        index='trigram', 
        required=True, 
    )
    
    #parent_path = fields.Char(index=True, unaccent=False)
    cat_description = fields.Char(
        string='category description', 
        default='CAD files are versioned and have dependencies.'
    )
    track_version = fields.Boolean(string='track version', default=True)
    track_depends = fields.Boolean(string='track depends', default=True)

    #relational fields


class hp_category_property(models.Model):
    #base fields
    _name = 'hp.category.property'
    _description = 'hp category property'
    _inherit = 'hp.common.model'

    #fields
    
    #relational fields
    cat_id = fields.Many2one(
        comodel_name='hp.category', 
        string='category id',
    )
    prop_id = fields.Many2one(
        comodel_name='hp.property',
        string='property',
    )

File: HackPDM/Src/Odoo/OdooModels/Models/HpCategory.cs
//using static System.Net.Mime.MediaTypeNames;

// ReSharper disable InconsistentNaming
using System;

namespace HackPDM.Odoo.OdooModels.Models;

[OdooModel(OdooDefaults.HP_CATEGORY_NAME, OdooDefaults.HP_CATEGORY)]
public partial class HpCategory : HpBaseModel<HpCategory>
{
    internal readonly string[] UsualExcludedFields = [];
    [OdooField(OdooFieldType.Char)]
    public string name;
	[OdooField(OdooFieldType.Char)]
	public string cat_description;
	[OdooField(OdooFieldType.Boolean)]
	public bool track_version;
	[OdooField(OdooFieldType.Boolean)]
	public bool track_depends;
    public HpCategory() { }
    public HpCategory(
        string name,
        string catDescription = "CAD files are versioned and have dependencies",
        bool trackVersion = true,
        bool trackDepends = true)
    {
        this.name = name;
        this.cat_description = catDescription;
        this.track_version = trackVersion;
        this.track_depends = trackDepends;
    }
}
public partial class HpCategory : HpBaseModel<HpCategory>
{
    public override string ToString()
    {
        return name;
    }
}